src/background.js

```diff
@@ -444,60 +422,61 @@ const rollout = {
     let skipHeuristicsCheck = await rollout.getSetting(DOH_SKIP_HEURISTICS_PREF, false);
     log("skipHeuristicsCheck: ", skipHeuristicsCheck);
 
-    if (!skipHeuristicsCheck) {
-      let shouldRunHeuristics = await stateManager.shouldRunHeuristics();
-      if (shouldRunHeuristics) {
-        await rollout.main();
-      }
-    }
+    await rollout.runHeuristics("startup");
 
     // Listen for network change events to run heuristics again
-    browser.experiments.netChange.onConnectionChanged.addListener(async () => {
+    browser.networkStatus.onConnectionChanged.addListener(async () => {
       log("onConnectionChanged");
-      // Only run the heuristics if user hasn't explicitly enabled/disabled DoH
-      let shouldRunHeuristics = await stateManager.shouldRunHeuristics();
-      let shouldShowDoorhanger = await stateManager.shouldShowDoorhanger();
-
-      if (shouldRunHeuristics) {
-        const netChangeDecision = await rollout.heuristics("netChange");
-        if (netChangeDecision === "disable_doh") {
-          await stateManager.setState("disabled");
-        } else if (shouldShowDoorhanger) {
-          await stateManager.showDoorHangerAndEnableDoH();
-        } else {
-          await stateManager.setState("enabled");
+
+      let linkInfo = await browser.networkStatus.getLinkInfo();
+      if (linkInfo.status !== "up") {
+        log("Link down.");
+        if (rollout.networkSettledTimeout) {
+          log("Canceling queued heuristics run.");
+          clearTimeout(rollout.networkSettledTimeout);
+          rollout.networkSettledTimeout = null;
         }
+        return;
       }
+
+      log("Queing a heuristics run in 60s, will cancel if network fluctuates.");
+      rollout.networkSettledTimeout = setTimeout(async () => {
+        log("No network fluctuation for 60 seconds, running heuristics.");
+        // Only run the heuristics if user hasn't explicitly enabled/disabled DoH
+        let shouldRunHeuristics = await stateManager.shouldRunHeuristics();
+
+        if (!shouldRunHeuristics) {
+          return;
+        }
+
+        // Only run the heuristics if user hasn't explicitly enabled/disabled DoH
+        rollout.runHeuristics("netChange");
```

Comment:
We need to set `rollout.networkSettledTimeout = null;` here as well (no need to `clearTimeout()` though since we're in the callback already.)